samigo/samigo-app/src/webapp/js/saveForm.js

```diff
@@ -143,6 +144,38 @@ function SaveFormContentAsync(toUrl, formId, buttonName, updateVar, updateVar2,
 	}
 	window.status = "";
 
+    //check noLateSubmission or isRetracted controlled by pastDueDate()
+    var noLateSubmission = text.indexOf("noLateSubmission");
+    var isRetracted = text.indexOf("isRetracted");
+    if (noLateSubmission >= 0 || isRetracted >= 0) {
+        timeExpired();
+        $("[id$=\\:submitNoCheck]")[0].click();
+    }
+
+    if (d != -1) {
```

Comment:
It's generally recommended to use `!==` rather than `!=`, to avoid coercion.

Reply:
Done